imap_processing/codice/codice_l1b.py

```diff
@@ -58,41 +56,42 @@ def convert_to_rates(
         "lo-sw-priority",
         "lo-ialirt",
     ]:
-        # Applying rate calculation described in section 10.2 of the algorithm
-        # document
-        # In order to divide by acquisition times, we must reshape the acq
-        # time data array to match the data variable shape
-        dims = [1] * dataset[variable_name].data.ndim
-        dims[1] = 128
-        acq_times = dataset.acquisition_time_per_step.data.reshape(dims)  # (128)
-        # Now perform the calculation
-        rates_data = dataset[variable_name].data / (
+        # Solve denominator, Acquisition times, of above products. Acquisition times
+        # must reshape to match the data variable shape
+        acq_times = dataset.acquisition_time_per_step.data[np.newaxis, :, np.newaxis]
```

Comment:
Can you stay in the proper xarray dimensions here and let the xarray broadcasting work itself out? If your acquisition_time_per_step has dimension `esa_step`, then it should broadcast out across the other dimensions automatically when doing the later multiplications. This has simplified a lot of the work on other instruments to try and stay in xarray-land rather than switch over to numpy axes.